components/content-service/pkg/layer/provider.go

```diff
@@ -523,6 +547,22 @@ func contentDescriptorToLayerPVC(cdesc []byte) (*Layer, error) {
 	return layerFromContent(layers...)
 }
 
+func workspaceReadyLayerPVC(src csapi.WorkspaceInitSource) (*Layer, error) {
+	msg := csapi.WorkspaceReadyMessage{
+		Source: src,
+	}
+	ctnt, err := json.Marshal(msg)
+	if err != nil {
+		return nil, err
+	}
+
+	return layerFromContent(
+		fileInLayer{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
+		fileInLayer{&tar.Header{Typeflag: tar.TypeDir, Name: "/.workspace/.gitpod", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755}, nil},
+		fileInLayer{&tar.Header{Typeflag: tar.TypeReg, Name: "/.workspace/.gitpod/ready", Uid: initializer.GitpodUID, Gid: initializer.GitpodGID, Mode: 0755, Size: int64(len(ctnt))}, []byte(ctnt)},
```

Comment:
this allows us to place ready file. 
why we cannot use `/workspace` folder? because when using pvc that folder is masked by pvc. 
also, on top of that, when restoring from prebuild for example, prebuild volume snapshot will contain ready file in there from the time when prebuild was generated. 
so this allows us to completely separate pvc and non pvc approach.